src/ui/Analytics/Analytics.ts

```diff
@@ -371,6 +373,42 @@ export class Analytics extends Component {
     this.client.setOriginContext(originContext);
   }
 
+  /**
+   * Re-enables analytics if they were previously disabled.
+   */
+  public enable() {
+    if (!this.disabled) {
+      console.warn('Analytics are already enabled.');
+      return;
+    }
+    super.enable();
+    this.initializeAnalyticsClient();
+    this.resolveQueryController().historyStore = new HistoryStore();
```

Comment:
Humm, I think this logic here should probably be moved to the query controller ?

So that it can determines internally if it needs to re-create the store or not, instead of having this driven by an external component.

Reply:
Good point, I'm moving it over to a new function. Since the store currently used by the [HistoryStore](/coveo/coveo.analytics.js/tree/master/src/history.ts) can't be determined publicly, I'm leaving it for now to just instantiating a new one. It should be easy to change later if needed.